truss/templates/triton/root/Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM nvcr.io/nvidia/tritonserver:23.07-py3
+
+RUN pip install --upgrade pip --no-cache-dir \
+    && rm -rf /root/.cache/pip
+
+ENV PYTHONUNBUFFERED True
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt update && \
+    apt install -y bash \
+                build-essential \
+                git \
+                curl \
+                ca-certificates \
+                software-properties-common \
+                nginx \
+                supervisor \
+                curl \
+    && apt-get autoremove -y \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY ./proxy.conf /etc/nginx/conf.d/proxy.conf
```

Comment:
Let's move all the copies in the end. Better to do all the heavy installs first for caching performance